pkg/kubelet/kubelet.go

```diff
@@ -1311,7 +1311,14 @@ func (kl *Kubelet) syncPod(o syncPodOptions) error {
 	kl.statusManager.SetPodStatus(pod, apiPodStatus)
 
 	// Kill pod if it should not be running
-	if errOuter := canRunPod(pod); errOuter != nil || pod.DeletionTimestamp != nil || apiPodStatus.Phase == api.PodFailed {
+	errOuter := canRunPod(pod)
+	if errOuter == nil {
+		rs := kl.runtimeState.networkErrors()
+		if len(rs) != 0 && !podUsesHostNetwork(pod) {
+			errOuter = fmt.Errorf("Network is not ready: %v", rs)
```

Comment:
Most of the pod that kubelet determined not to run would be rejected at admission, and not entering syncpod at all.